tests/test_ai_synthesis_pass.py

```diff
@@ -0,0 +1,396 @@
+# -*- coding: utf-8 -*-
+
+# (C) Copyright 2024 IBM. All Rights Reserved.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for the ai synthesis pass"""
+
+import pytest
+
+from qiskit import QuantumCircuit
+
+from qiskit.transpiler import PassManager
+
+from qiskit_ibm_transpiler.ai.collection import (
+    CollectPermutations,
+    CollectLinearFunctions,
+    CollectCliffords,
+    CollectPauliNetworks,
+)
+from qiskit_ibm_transpiler.ai.synthesis import (
+    AIPermutationSynthesis,
+    AILinearFunctionSynthesis,
+    AICliffordSynthesis,
+    AIPauliNetworkSynthesis,
+)
+
+from tests.parametrize_functions import (
+    parametrize_local_mode,
+    parametrize_coupling_map_format,
+)
+
+
+# TODO: For Permutations, the original circuit doesn't return a DAGCircuit with nodes. Decide how the code should behave on this case
+def parametrize_basic_circuit_collector_pass_and_ai_synthesis_pass():
```

Comment:
This is probably because the perm circuit is very small. It only has 3 qubits and 2 permutations. We would need to define a circuit which triggers the collection + synthesis.